gh-77065: Add optional keyword-only argument `echo_char` for `getpass.getpass` (#130496)
python/cpython: Doc/library/getpass.rst

The :mod:`getpass` module provides two functions:

> .. function:: getpass(prompt='Password: ', stream=None, mask=None)

Reviewer: ```suggestion
.. function:: getpass(prompt='Password: ', stream=None, *, echochar=None)
```

> .. function:: getpass(prompt='Password: ', stream=None, *, echochar=None)

Reviewer 2: Let's add an underscore throughout for readability and accessibility:

```suggestion
.. function:: getpass(prompt='Password: ', stream=None, *, echo_char=None)
```

Author: thank you. left `echochar` only in function names for tests

Reviewer 2: Let's update those too: we need to read them as well, and if we search for the name, they will also show up.

Author: done

   (:file:`/dev/tty`) or if that is unavailable to ``sys.stderr`` (this
   argument is ignored on Windows).

>    The *echochar* argument controls how user input is displayed while typing.

Reviewer: Let's also indicate that `echochar` should be an ASCII character.

   If *echochar* is ``None`` (default), input remains hidden. Otherwise,
   *echochar* must be a printable ASCII string and each typed character

>    is replaced by the former. For example, ``echochar='*'`` will display

Reviewer 2: ```suggestion
   is replaced by it. For example, ``echochar='*'`` will display
```

>    input.

Reviewer: You should add an `.. versionchanged:: next` with a note like "Add the *echochar* parameter for keyboard feedback."

>    If *echochar* is ``None`` (default), input remains hidden. If *echochar* is
>    a string, each typed character is replaced with the given string. But this
>    string must be ASCII character. For example, ``echochar='*'`` will display
>    asterisks instead of the actual input.

Reviewer: ```suggestion
   If *echochar* is ``None`` (default), input remains hidden. Otherwise,
   *echochar* must be a printable ASCII string and each typed character
   is replaced by the former. For example, ``echochar='*'`` will display
   asterisks instead of the actual input.
```
